lib/jormungandr/src/Cardano/Wallet/Binary/Jormungandr.hs

```diff
@@ -141,14 +152,85 @@ getInitial = do
     len <- fromIntegral <$> getWord16be
     replicateM len getConfigParam
 
+getTransaction :: Get Tx
+getTransaction = isolate 43 $ do
+    (ins, outs) <- getTokenTransfer
+
+    let witnessCount = length ins
+    _wits <- replicateM witnessCount getWitness
+
+    return $ Tx ins outs
+  where
+    getWitness = do
+        tag <- getWord8
+        case tag of
+            1 -> isolate 128 $ do
+                -- Old address witness scheme
+                xpub <- getByteString 64
+                sig <- Hash <$> getByteString 64
+                return $ PublicKeyWitness xpub sig
+
+            2 -> isolate 64 $ do
+                _sig <- Hash <$> getByteString 64
+                error "unimplemented: New address witness scheme"
+
+            3 -> isolate 68 $ do
+                error "unimplemented: Account witness"
+            other -> fail $ "Invalid witness type: " ++ show other
+
+
+{-------------------------------------------------------------------------------
+                            Common Structure
+-------------------------------------------------------------------------------}
+
+getTokenTransfer :: Get ([TxIn], [TxOut])
+getTokenTransfer = do
+    inCount <- fromIntegral <$> getWord8
+    outCount <- fromIntegral <$> getWord8
+    ins <- replicateM inCount getInput
+    outs <- replicateM outCount getOutput
+    return (ins, outs)
+  where
+    getInput = isolate 41 $ do
+        -- NOTE: special value 0xff indicates account spending
+        index <- fromIntegral <$> getWord8
+        tx <- Hash <$> getByteString 32
+
+        return $ TxIn tx index
+
+    getOutput = do
+        addr <- getAddress
+        value <- Coin <$> getWord64be
+        return $ TxOut addr value
+
+    getAddress = do
+        headerByte <- getWord8
+        let kind = kindValue headerByte
+        let _discrimination = discriminationValue headerByte
+        case kind of
+            -- Single Address
+            0x3 -> Address <$> getByteString 32
```

Comment:
where can I find this info? I can't seem to find it in https://github.com/input-output-hk/rust-cardano/blob/master/chain-impl-mockchain/doc/format.md#preliminaries

Reply:
Should be from here https://github.com/input-output-hk/rust-cardano/blob/e0616f13bebd6b908320bddb1c1502dea0d3305a/chain-addr/src/lib.rs#L392-L443